### PR TITLE
Docs for VideoEncoder.isConfigSupported()

### PR DESCRIPTION
**Pull request by Djuffin**
#### Summary
Adding docs for VideoEncoder.isConfigSupported()

#### Motivation
Undocumented method 

#### Supporting details
https://www.w3.org/TR/webcodecs/#dom-videoencoder-isconfigsupported

#### Metadata
This PR…
- [x ] Adds a new document

### CODE REVIEW

File: files/en-us/web/api/videoencoder/index.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ The **`VideoEncoder`** interface of the {{domxref('WebCodecs API','','','true')}
   - : Enqueues a control message to encode a given {{domxref("VideoFrame")}}.
 - {{domxref("VideoEncoder.flush()")}}
   - : Returns a promise that resolves once all pending messages in the queue have been completed.
+- {{domxref("VideoEncoder.isConfigSupported()")}}
+  - : Returns a promise indicating whether the provided VideoEncoderConfig is supported.
```
Comment (sideshowbarker) [inline files/en-us/web/api/videoencoder/index.md:36]:
```suggestion
  - : Returns a promise indicating whether the provided `VideoEncoderConfig` is supported.
```

File: files/en-us/web/api/videoencoder/isconfigsupported/index.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  - Reference
+  - isConfigSupported
+  - VideoEncoder
+browser-compat: api.VideoEncoder.isConfigSupported
+---
+{{securecontext_header}}{{DefaultAPISidebar("WebCodecs API")}}
+
+The **`isConfigSupported()`** static method of the {{domxref("VideoEncoder")}} interface that checks if the given config is supported.
+
+That is, if it can successfully configure {{domxref("VideoEncoder")}} objects with this the same config.
```
Comment (sideshowbarker) [inline files/en-us/web/api/videoencoder/isconfigsupported/index.md:17]:
```suggestion
The **`isConfigSupported()`** static method of the {{domxref("VideoEncoder")}} interface checks if the given config is supported (that is, if {{domxref("VideoEncoder")}} objects can be successfully configured with the given config.
```

File: files/en-us/web/api/videoencoder/isconfigsupported/index.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  - API
+  - Method
+  - Reference
+  - isConfigSupported
+  - VideoEncoder
+browser-compat: api.VideoEncoder.isConfigSupported
+---
+{{securecontext_header}}{{DefaultAPISidebar("WebCodecs API")}}
+
+The **`isConfigSupported()`** static method of the {{domxref("VideoEncoder")}} interface checks if the given config is supported (that is, if {{domxref("VideoEncoder")}} objects can be successfully configured with the given config.
```
Comment (sideshowbarker) [inline files/en-us/web/api/videoencoder/isconfigsupported/index.md:15]:
```suggestion
The **`isConfigSupported()`** static method of the {{domxref("VideoEncoder")}} interface checks if the given config is supported (that is, if {{domxref("VideoEncoder")}} objects can be successfully configured with the given config).
```

File: files/en-us/web/api/videoencoder/isconfigsupported/index.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+### Parameters
+
+- `config`
+  - : The dictionary object accepted by {{jsxref("VideoEncoder.configure")}}
+
+### Return value
+
+A {{jsxref("Promise")}} that resolves with an object containing the following members:
+  - `supported`{{Optional_Inline}}
+    - : A Boolean value which is `true` if the given config is supported by the encoder.
```
Comment (sideshowbarker) [inline files/en-us/web/api/videoencoder/isconfigsupported/index.md:32]:
```suggestion
    - : A boolean value which is `true` if the given config is supported by the encoder.
```

File: files/en-us/web/api/videoencoder/isconfigsupported/index.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+- `config`
+  - : The dictionary object accepted by {{jsxref("VideoEncoder.configure")}}
+
+### Return value
+
+A {{jsxref("Promise")}} that resolves with an object containing the following members:
+  - `supported`{{Optional_Inline}}
+    - : A boolean value which is `true` if the given config is supported by the encoder.
+  - `config`{{Optional_Inline}}
+    - : An copy of the given config with all the field recognized by the encoder.
```
Comment (sideshowbarker) [inline files/en-us/web/api/videoencoder/isconfigsupported/index.md:34]:
```suggestion
    - : A copy of the given config with all the field recognized by the encoder.
```

File: files/en-us/web/api/videoencoder/isconfigsupported/index.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+- `config`
+  - : The dictionary object accepted by {{jsxref("VideoEncoder.configure")}}
+
+### Return value
+
+A {{jsxref("Promise")}} that resolves with an object containing the following members:
+  - `supported`{{Optional_Inline}}
+    - : A boolean value which is `true` if the given config is supported by the encoder.
+  - `config`{{Optional_Inline}}
+    - : A copy of the given config with all the field recognized by the encoder.
```
Comment (sideshowbarker) [inline files/en-us/web/api/videoencoder/isconfigsupported/index.md:34]:
```suggestion
    - : A copy of the given config with all the fields recognized by the encoder.
```

File: files/en-us/web/api/videoencoder/isconfigsupported/index.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    - : A boolean value which is `true` if the given config is supported by the encoder.
+  - `config`{{Optional_Inline}}
+    - : A copy of the given config with all the fields recognized by the encoder.
+
+### Exceptions
+
+- {{jsxref("TypeError")}}
+  - : Thrown if the provided `config` is invalid, i.e. doesn't have required
+  values (such as an empty `codec` filed) or has invalid values (such as a
+  negative `width`)
```
Comment (sideshowbarker) [inline files/en-us/web/api/videoencoder/isconfigsupported/index.md:41]:
```suggestion
  - : Thrown if the provided `config` is invalid; that is, if doesn't have required
  values (such as an empty `codec` field) or has invalid values (such as a
  negative `width`)
```

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/api/videoencoder/isconfigsupported/index.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+## Syntax
+
+```js
+isConfigSupported(config)
+```
+
+### Parameters
+
+- `config`
+  - : The dictionary object accepted by {{jsxref("VideoEncoder.configure")}}
```
Comment (sideshowbarker) [inline files/en-us/web/api/videoencoder/isconfigsupported/index.md:26]:
```suggestion
  - : The dictionary object accepted by the {{domxref("VideoEncoder.configure")}}
```

**Comment by sideshowbarker** [comment]
Eugene, thanks much, and congrats on landing your first docs change here — welcome aboard 🎉

**Comment by Djuffin** [comment]
@sideshowbarker Thank you so much for quick editing and turnaround time!

**Pull request merged**